sdk/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonIUD.java

```diff
@@ -173,10 +184,16 @@ public void update(String path, Expression filterExpression,
     RecordWriter<NullWritable, ObjectArrayWritable> deleteDeltaWriter =
         CarbonTableOutputFormat.getDeleteDeltaRecordWriter(path);
     ObjectArrayWritable writable = new ObjectArrayWritable();
-
+    long day = 24L * 3600 * 1000;
     while (reader.hasNext()) {
       Object[] row = (Object[]) reader.readNextRow();
       writable.set(Arrays.copyOfRange(row, row.length - 1, row.length));
+      for (Integer dateIndex : dateIndexes) {
+        row[dateIndex] = new Date((day * ((int) row[dateIndex])));
```

Comment:
Date and Timestamp conversion should be in SDK reader. Please fix the reader

Reply:
Done